pydoctor/templatewriter/summary.py

```diff
@@ -26,13 +26,26 @@ def moduleSummary(module: model.Module, page_url: str) -> Tag:
         r(class_='private')
     if not isinstance(module, model.Package):
         return r
-    contents = [m for m in module.contents.values()
-                if isinstance(m, model.Module) and m.isVisible]
+    contents = list(module.submodules())
     if not contents:
         return r
-    ul = tags.ul()
+
     def fullName(obj: model.Documentable) -> str:
         return obj.fullName()
+
+    if len(contents) > 50 and not any(any(s.submodules()) for s in contents):
```

Comment:
What if there is 50 sub modules and one package? 
I think the 50 du modules should still be grouped together while the packages would be extended.

Also packages are instances of model.Package, so maybe it could work with an `isinstance` check instead ?

Reply:
I believe that having more than 50 submodules and one package is really uncommon. I don't think I have ever seen that in a Python package, have you? If not I'd rather keep it simple.

Reply:
Very well